modules/ROOT/pages/consume-data-from-an-api.adoc

```diff
@@ -0,0 +1,323 @@
+
+* Authentication
+* A base URI for configuring the outbound HTTP endpoint
+* Scope
++
+Based on authentication credentials, the application can limit calls to only GET requests on certain resources, for example, or otherwise limit access to a resource.
++
+* Resources
++
+You must configure the path of the resources for the outbound HTTP endpoint.
++
+* Methods
++
+You can accept POST as the default method for the HTTP outbound endpoint, or you can change this default to the method supported by your resource.
++
+* Input validation
++
+For POST and PUT calls, the API almost always requires that the input format, such as JSON or XML, match its schema. +
+
+In Mule runtime engine, you can define or configure the following:
+
+* Output format
++
+After the HTTP request operation, your flow must include message processors to accept the response from the API and translate the response into a useable format.
++
+* Redirects
++
+If the API uses redirects, enable redirects in your HTTP request operation.
++
+* Timeout
+* Headers for methods
++
+POST, PUT, and DELETE requests almost always require headers.
++
+* URI and Query Parameters
+* Error handling
+
+See xref:connectors::http-connector[HTTP Connector documentation] for more information about how to configure the request operation.
+
+
+=== Consume REST API Example
+
+The following example consumes the https://jsonplaceholder.typicode.com/todos/1[JSONPlaceholder API]. The example shows only the configuration of the HTTP Request Connector, and to make it work, you must configure at least a  message source to trigger a message in the flow and at least one message processor to accept the response. See xref:rest-api-examples.adoc[REST API Examples documentation] for more information.
+
+In this API example:
+
+* The API doesn't require authentication.
+* The base URL is `jsonplaceholder.typicode.com`.
+* The API is a mock API for development purposes.
+* The sample API requests show that to get information about user with `id=1` the request must be:
+
+----
+https://jsonplaceholder.typicode.com/users/1
+----
+
+The parts of the URL are as follows:
+
+* Protocol: `http`
+* API Base URL: `jsonplaceholder.typicode.com`
+* Resource path: `/users/1`
+* Query Parameters: None, in this case.
+
+To implement this call in your Mule application, configure your HTTP request operation.
+
+=== Configuration HTTP Request Operation in Anypoint Studio
+
+. From the Mule Palette in Studio, add an HTTP Request Connector to your application.
+. In the HTTP Request Configuration window, provide `jsonplaceholder.typicode.com` as the value for *Host*.
+. Set the *Port* value to 80.
+. Click *OK*:
+
+.HTTP Request configuration with Host and Port.
+image::mruntime-consume-rest-api-configuration-1.png[HTTP Request configuration with Host and Port]
+
+[start=5]
+. Navigate to the Request section from the HTTP Request operation’s properties editor > General tab and add `/users/1` in the *Path* field. +
+
```

Comment:
delete line 120 so that the next sentence is at the same level as point number 5.